Gems/Atom/Feature/Common/Code/Source/Mesh/MeshFeatureProcessor.cpp

```diff
@@ -1026,6 +1022,118 @@ namespace AZ
+                        // (approximately) area-preserving to get a properly weighted average; and
+                        // mostly, (3) it assumes that a single 'average color' is a meaningful
+                        // characterisation of the full material.
+                        Color avgColor = baseColorStreamingImg->GetAverageColor();
+
+                        // We do a simple 'multiply' blend with the base color for now. Warn
+                        // the user if something else was intended.
+                        propertyIndex = material->FindPropertyIndex(AZ::Name("baseColor.textureBlendMode"));
+                        if (propertyIndex.IsValid())
+                        {
+                            AZ::Name textureBlendMode = material->GetMaterialPropertiesLayout()
+                                     ->GetPropertyDescriptor(propertyIndex)
+                                     ->GetEnumName(material->GetPropertyValue<uint32_t>(propertyIndex));
+                            if (textureBlendMode != AZ::Name("Multiply"))
+                            {
+                                AZ_Warning("MeshFeatureProcessor", false, "textureBlendMode '%s' is not "
+                                        "yet supported when requesting BaseColor irradiance source, "
+                                        "defaulting to 'Multiply'.", textureBlendMode.GetCStr());
+                            }
+                        }
+                        // 'Multiply' blend mode:
+                        subMesh.m_irradianceColor = avgColor * subMesh.m_baseColor;
+                    }
+                    else
+                    {
+                        AZ_Warning("MeshFeatureProcessor", false, "Using BaseColor as irradianceColorSource "
+                                "is currently only supported for textures of type StreamingImage");
+                        // Default to the flat base color
+                        subMesh.m_irradianceColor = subMesh.m_baseColor;
+                    }
+                }
+                else
+                {
+                    // No texture, simply copy the baseColor
+                    subMesh.m_irradianceColor = subMesh.m_baseColor;
+                }
+            }
+            else if (irradianceColorSource == AZ::Name("Manual"))
+            {
+                propertyIndex = material->FindPropertyIndex(AZ::Name("irradiance.manualColor"));
+                if (propertyIndex.IsValid())
+                {
+                    subMesh.m_irradianceColor = material->GetPropertyValue<AZ::Color>(propertyIndex);
+                }
+                else
+                {
+                    AZ_Warning("MeshFeatureProcessor", false, "Requested manual color as irradianceColorSource "
+                            "but I could not find an irradiance.manualColor field. Defaulting to black.");
+                    subMesh.m_irradianceColor = AZ::Color(0.0f);
+                }
```

Comment:
I think we should also look for "irradiance.color" as a fallback in case there is some other material type that does not have the automatic concept at all. That will allow that material type the option of using just "irradiance.color" for the name.